cmd/krel/cmd/release_notes.go

```diff
@@ -272,9 +284,69 @@ func runReleaseNotes() (err error) {
 	return nil
 }
 
+// validateDraftPRRepository checks if a repo is fit to create a PR
+func validateDraftPRRepository() error {
+	gh := github.New()
+
+	isrepo, err := gh.RepoIsForkOf(
+		releaseNotesOpts.draftOrg, releaseNotesOpts.draftRepo, defaultKubernetesSigReleaseOrg, defaultKubernetesSigReleaseRepo,
+	)
+	if err != nil {
+		return errors.Wrapf(
+			err, "while checking if repository is a fork of %s/%s",
+			defaultKubernetesSigReleaseOrg, defaultKubernetesSigReleaseRepo,
+		)
+	}
+
+	if !isrepo {
+		return errors.New(
+			fmt.Sprintf(
+				"Cannot create PR, %s/%s is not a fork of %s/%s",
+				releaseNotesOpts.draftOrg, releaseNotesOpts.draftRepo,
+				defaultKubernetesSigReleaseOrg, defaultKubernetesSigReleaseRepo,
+			),
+		)
+	}
+
+	return nil
+}
+
+// validateWebsitePRRepository checks if a repo is fit to create a PR
+func validateWebsitePRRepository() error {
+	gh := github.New()
+
+	isrepo, err := gh.RepoIsForkOf(
+		releaseNotesOpts.websiteOrg, releaseNotesOpts.websiteRepo,
+		defaultKubernetesSigsOrg, defaultKubernetesSigsRepo,
+	)
+	if err != nil {
+		return errors.Wrapf(
+			err, "while checking if repository is a fork of %s/%s",
+			defaultKubernetesSigsOrg, defaultKubernetesSigsRepo,
+		)
+	}
+
+	if !isrepo {
+		return errors.New(
+			fmt.Sprintf(
+				"Cannot create PR, %s/%s is not a fork of %s/%s",
+				releaseNotesOpts.websiteOrg, releaseNotesOpts.websiteRepo,
+				defaultKubernetesSigsOrg, defaultKubernetesSigsRepo,
+			),
+		)
+	}
+
+	return nil
+}
+
 // validateDraftPROptions checks if we have all needed parameters to create the Release Notes PR
 func validateDraftPROptions() error {
 	if releaseNotesOpts.createDraftPR {
+		token, isset := os.LookupEnv("GITHUB_TOKEN")
+		if !isset || token == "" {
+			return errors.New("Cannot create release notes draft if GITHUB_TOKEN is not set")
+		}
```

Comment:
Hm, either we add it as public utility function to `pkg/github/github.go` or we just use `github.TokenEnvKey`   here.

Reply:
Switched to `github.TokenEnvKey` although i think we should add a new function to ensure an authenticated client. I'll add it tomorrow.